src/omni-router/omni-router.ts

```diff
@@ -1,36 +1,61 @@
 import { BigNumber, BigNumberish, ContractTransaction } from 'ethers'
+import { BurningQueue as BurningQueueInterface } from 'typechain/BurningQueue'
+import { PromiseOrValue } from 'typechain/common'
+import { IIndexViewer } from 'typechain/OmniIndex'
+import { SubIndexLib } from 'typechain/SubIndexFactory'
+import { Address } from 'types'
 
-import { Account } from '../account'
-import { Contract } from '../contract'
-import {
-  OmniRouter as OmniRouterInterface,
-  OmniRouter__factory,
-} from '../typechain'
-import { PromiseOrValue } from '../typechain/common'
-import { Address, ChainId, ChainIds } from '../types'
+import { Zero0xQuoteOptions, ZeroExAggregator } from '../0x-aggregator'
 
-import {
-  defaultOmniMessageRouterAddress,
-  OmniMessageRouter,
-} from './omni-message-router'
+import { createQuotes, createRemoteBatches } from './batches-utils'
+import { BurningQueue } from './burning-queue'
+import { OmniIndex } from './omni-index'
+import { OmniRouterInterface } from './omni-router-types'
+import { RedeemRouter } from './redeem-router'
+import { SubIndex } from './sub-index-factory'
 
-/** ### Default OmniRouter address for network */
-export const defaultOmniRouterAddress: Record<ChainId, Address> = {
-  /** ### Default OmniRouter address on goerli rollup testnet. */
-  [ChainIds.GoerliRollupTestnet]: '0xbce372299dab3c06287fee4b313a082181ebe96c',
-}
-
-export class OmniRouter extends Contract<OmniRouterInterface> {
+/** ### OmniRouter class */
+export class OmniRouter implements OmniRouterInterface {
   /**
    * ### Creates a new OmniRouter instance
    *
-   * @param account Account to use for signing
-   * @param contract Contract instance or address of the OmniRouterInterface contract
+   * @param omniIndex instance of OmniIndex
+   * @param redeemRouter instance of RedeemRouter
+   * @param subIndex instance of SubIndex
+   * @param burningQueue instance of BurningQueue
+   * @param zeroExAggregator ZeroEx client
    *
-   * @returns New OmniRouterInterface token instance
+   * @returns New OmniRouter instance
+   */
+  constructor(
+    public readonly omniIndex: OmniIndex,
```

Comment:
I cannot understand the reason to have DYNAMIC param `index` as a constructor variable

Reply:
TODO added